compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala

```diff
@@ -38,6 +38,8 @@ class YCheckPositions extends Phase {
             // Recursivlely check children while keeping track of current source
             reporting.trace(i"check pos ${tree.getClass} ${tree.source} ${sources.head} $tree") {
               tree match {
+                case tree: TypeDef if tree.isClassDef && MacroAnnotations.hasMacroAnnotation(tree.symbol) =>
```

Comment:
We should probably do this for any `DefTree` that has a macro annotation.

Reply:
Good point, I added this behavior for both `DefDef`s and `ValDef`s, and added both to the test